libs/hdf-converters/src/splunk-mapper.ts

```diff
@@ -303,33 +299,69 @@ export class SplunkMapper {
   }
 
   async queryData(query: string): Promise<any[]> {
+    // Request session key for Axios instance
+    const authToken = await checkSplunkCredentials(this.config);
+    this.axiosInstance.defaults.headers.common[
+      'Authorization'
+    ] = `Bearer ${authToken}`;
+
+    // Create new search job from given query
     const job = await this.createJob(query);
+
     return new Promise((resolve, reject) => {
-      job.track(
-        {},
-        {
-          done: () => {
-            logger.debug(`Getting results for query: ${query}`);
-            job.results({count: 100000}, (err, results) => {
-              if (err) {
-                reject(err);
-              } else {
-                try {
-                  resolve(this.parseSplunkResponse(query, results));
-                } catch (e) {
-                  reject(e);
-                }
+      // Ping Splunk instance every 50 ms on status of search job
+      const awaitJob = setInterval(() => {
+        this.axiosInstance
+          .get(`${this.hostname}/services/search/jobs/${job}`)
+          .then(
+            async (response) => {
+              // If search job is complete, kill interval loop and ping Splunk for search job results
+              if (
+                response.data.entry[0].content.dispatchState == 'DONE' &&
```

Comment:
I'll take your word that it returns an array since that doesn't seem to come through clearly from the XML version of the output.  If it's returning an array, I guess I'm just unsure why it's only an array of length 1?  Is there a way to force it to return an array of length larger than 1?  If we get it to be a larger array, is it possible for the dispatchState to be 'done' in some but not all of them (which could be the source of a possible error)?

In any case, it can't hurt to do defensive ~~driving~~ parsing.

Reply:
It appears that the array is a quirk of how Splunk responds to this request when asking for a json formatted response. The SDK doesn't indicate that there are multiple fields in `entry` and the docs don't show any parameters which could be used to prompt such. 

Just in case, I put in some handling to ensure that we're always using the first entry item and to throw an error if there are every any instance of hacing >1 or <1 items in the array since that's unexpected.